benchexec/tools/ultimate.py

```diff
@@ -321,18 +310,12 @@ def program_files(self, executable):
         )
         return [executable] + self._program_files_from_executable(executable, paths)
 
-    def determine_result(self, returncode, returnsignal, output, is_timeout):
-        if self._uses_propertyfile:
-            return self._determine_result_with_propertyfile(
-                returncode, returnsignal, output, is_timeout
-            )
-        return self._determine_result_without_propertyfile(
-            returncode, returnsignal, output, is_timeout
-        )
+    def determine_result(self, run):
+        if any([arg for arg in run.commandline if "--spec" in arg or ".prp" in arg]):
```

Comment:
```suggestion
        if any("--spec" in arg or ".prp" in arg for arg in run.commandline):
```

But do you actually want to the check whether there is one argument that contains `--spec`, or whether there is one argument that is equal to `--spec`?
If the latter:
```suggestion
        if any(arg == "--spec" or ".prp" in arg for arg in run.commandline):
```